### PR TITLE
Perfect json content

### PR DESCRIPTION
**Pull request by q384566678**
Signed-off-by: zhouhao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

File: specs-go/config.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ type Spec struct {
 	// Mounts configures additional mounts (on top of Root).
 	Mounts []Mount `json:"mounts,omitempty"`
 	// Hooks configures callbacks for container lifecycle events.
-	Hooks Hooks `json:"hooks"`
+	Hooks Hooks `json:"hooks,omitempty"`
```
Comment (wking) [inline specs-go/config.go]: This is covered in #427.

File: specs-go/state.go [hunk 2]
```diff
@@ -13,5 +13,5 @@ type State struct {
 	// BundlePath is the path to the container's bundle directory.
 	BundlePath string `json:"bundlePath"`
 	// Annotations are the annotations associated with the container.
-	Annotations map[string]string `json:"annotations"`
+	Annotations map[string]string `json:"annotations,omitempty"`
```
Comment (wking) [inline specs-go/state.go]: The in-flight #507 is adding the associated REQUIRED and OPTIONAL wording to `runtime.md`.

File: schema/config-linux.json [hunk 3]
```diff
@@ -89,7 +89,10 @@
                                 "id": "https://opencontainers.org/schema/bundle/linux/resources/pids/limit",
                                 "$ref": "defs.json#/definitions/int64"
                             }
-                        }
+                        },
+			"required": [
```
Comment (wking) [inline schema/config-linux.json]: Inconsistent indent.  Try running `make fmt` from the `schema` directory.

File: schema/config-schema.json [hunk 4]
```diff
@@ -97,6 +118,10 @@
                 "user": {
                     "id": "https://opencontainers.org/schema/bundle/process/user",
                     "type": "object",
+                    "required": [
+			"uid",
+			"gid"
```
Comment (wking) [inline schema/config-schema.json]: These are [only required on Linux/Solaris][1], and there's no way to enforce that constraint in JSON Schema without some serious `oneOf` hoop-jumping.

[1]: https://github.com/opencontainers/runtime-spec/blob/ced3365b4d5856371a61cb9b0f9c2e69c3163643/config.md#linux-and-solaris-user

File: schema/config-schema.json [hunk 5]
```diff
@@ -170,8 +195,6 @@
         "ociVersion",
         "platform",
         "process",
-        "root",
-        "mounts",
-        "hooks"
```
Comment (wking) [inline schema/config-schema.json]: Removing `hooks` is covered by the in-flight #427.

File: specs-go/state.go (hunk 2 above)
Comment (q384566678) [inline specs-go/state.go]: Since it is optional, it should not be so modified?

Comment (wking) [inline specs-go/state.go]: Adding `omitempty` (as you do here and as #507 does not do) makes sense.  Adding REQUIRED/OPTIONAL to the Markdown spec (as #507 does and you do not do here) also makes sense.  Once both PRs land, we'll be in the right place ;).

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**